Sources/Conditions/CreditCardCondition.swift

```diff
@@ -0,0 +1,102 @@
+//
+//  CreditCardCondition.swift
+//  FormValidatorSwift
+//
+//  Created by Onur Ersel on 2016-11-02.
+//  Copyright © 2016 ustwo. All rights reserved.
+//
+
+import Foundation
+
+
+/// Credit card types that are supported by `CreditCardCondition`. There is a `regex` property that returns the regex for validating that credit card number.
+public enum CreditCardType : String {
+    case americanExpress    = "American Express"
+    case dinersClub         = "Diners Club"
+    case discover           = "Discover"
+    case jcb                = "JCB"
+    case maestro            = "Maestro"
+    case mastercard         = "Master Card"
+    case visa               = "VISA"
+    case all                = "all"
+    
+    var regex: String {
+        switch self {
+        case .americanExpress:
+            return "3[47][0-9]{13}"
+        case .dinersClub:
+            return "3(?:0[0-5]|[68][0-9])[0-9]{11}"
+        case .discover:
+            return "6(?:011|5[0-9]{2})(?:[0-9]{12})"
+        case .jcb:
+            return "(?:2131|1800|35\\d{3})\\d{11}"
+        case .maestro:
+            return "(?:5[0678]\\d\\d|6304|6390|67\\d\\d)\\d{8,15}"
+        case .mastercard:
+            return "5[1-5][0-9]{14}"
+        case .visa:
+            return "^4[0-9]{12}(?:[0-9]{3})?$"
+        case .all:
```

Comment:
All can make sense in an `enum`, but I wouldn't use it as a case. This would not make logical sense in an enum:

```swift
let cards: [CreditCardType] = [.all, .visa]
```

Instead I'd make it a static property. An example would be:

```swift
public enum CreditCardType: String {
  case jcb
  case visa

  static let allTypes: [CreditCardType] = [.jcb, .visa]
}
```

I still think that OptionSet makes more sense, as you are typically going to be using this as a set of multiple values and the calculation to determine membership will be much faster with an OptionSet arithmetic rather than using `contains` on an array of enum values.